### PR TITLE
[Workflow] Allow using public properties for the marking store

### LINKED ISSUE #18656: [Workflow] Add support for storing the marking in a property
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#50974
| PR author(s) | @lyrixx
| Merged in    | 6.4

We created this issue to not forget to document this new feature. We would really appreciate if you can help us with this task. If you are not sure how to do it, please ask us and we will help you.

To fix this issue, please create a PR against the 6.4 branch in the [symfony-docs repository](https://github.com/symfony/symfony-docs).

Thank you! :smiley:

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix https://github.com/symfony/symfony-docs/issues/18656

### CODE REVIEW

**Comment by lyrixx** [comment]
Can you add a note about the context: Context is not supported with property.

So, if one want to use the context, they must use a public method.

But the following code works:
```php
class S
{

  public $marking; // Used for reading
  private $context;

  public function setMarking($marking, $context) { // used for setting 
     ...
  }
}
```

**Comment by alexandre-daubois** [comment]
Pushed a new version, what do you think? :slightly_smiling_face:

**Review by lyrixx — APPROVED** [review]
Perfect 💛 
Thank you so much for all your contributions 😍

File: workflow.rst [hunk 1]
```diff
@@ -190,12 +190,50 @@ The configured property will be used via its implemented getter/setter methods b
...
         }
 
-        public function setCurrentPlace($currentPlace, $context = []): void
+        public function setCurrentPlace(string $currentPlace, array $context = []): void
         {
             $this->currentPlace = $currentPlace;
         }
     }
 
+.. tip::
```
Comment (OskarStark) [inline workflow.rst]: I think we can use a regular paragraph here, it is too long for a tip IMHO

**Review by OskarStark — APPROVED** [review]

File: workflow.rst (hunk 1 above)
Comment (alexandre-daubois) [reply workflow.rst]: You're right, updated :slightly_smiling_face:

**Comment by javiereguiluz** [comment]
Another great combo of nice feature + nice contribution + nice review. Thank you all!

**Pull request merged**